bilby/core/sampler/__init__.py

```diff
@@ -251,6 +251,14 @@ def run_sampler(
     meta_data["likelihood"] = likelihood.meta_data
     meta_data["loaded_modules"] = loaded_modules_dict()
     meta_data["environment_packages"] = env_package_list(as_dataframe=True)
+    # Add the cosmology if available
```

Comment:
I agree that 'users didn't specify the result type' shouldn't be reason enough in principle, but I'm slightly concerned the documentation is sufficiently clear on this (AFAICT, only two of all our GW examples specify it). Therefore, anyone that's used one of the examples that doesn't have it as a starting point may not have specified the class. It's also different to e.g. the likelihood metadata which is always saved irrespective of the results class, meaning it can be recovered even if the wrong result class is used.

I do see your point, as this is clearly GW code. This may be out-of-scope for this MR, but I could imagine having an object that keeps track of some of the global variables, e.g. `rng`? This could then be saved in the metadata. If we had something like this, calling `set_cosmology` could just add the cosmology to that object?

Either way, I think we should update the GW examples to include the correct result class.

Reply:
So I've removed these lines and had a go at implementing a global meta data object in: https://github.com/bilby-dev/bilby/pull/873. I've also opened an issue to track updates to the examples: https://github.com/bilby-dev/bilby/pull/873

Reply:
I've also moved `cosmology` to the `global_meta_data` so it is consistent with #873